backend/src/apiserver/webhook/pipelineversion_webhook.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2025.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	"github.com/kubeflow/pipelines/backend/src/apiserver/common"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/template"
+	k8sapi "github.com/kubeflow/pipelines/backend/src/crd/kubernetes/v2beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	ctrladmission "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var scheme *runtime.Scheme
+
+func init() {
+	scheme = runtime.NewScheme()
+	err := k8sapi.AddToScheme(scheme)
+	if err != nil {
+		// Panic is okay here because it means there's a code issue and so the package shouldn't initialize.
+		panic(fmt.Sprintf("Failed to initialize the Kubernetes API scheme: %v", err))
+	}
+}
+
+type PipelineVersionsWebhook struct {
+	Client ctrlclient.Client
+}
+
+var _ ctrladmission.CustomValidator = &PipelineVersionsWebhook{}
+
+func newBadRequestError(msg string) *apierrors.StatusError {
+	return &apierrors.StatusError{
+		ErrStatus: metav1.Status{
+			Code:    http.StatusBadRequest,
+			Reason:  metav1.StatusReasonBadRequest,
+			Message: msg,
+		},
+	}
+}
+
+func (p *PipelineVersionsWebhook) ValidateCreate(
```

Comment:
I may have missed it, but I don't see this validation:

> Verify the spec.pipelineSpec.pipelineInformation.name field matches the metadata.name field

Reply:
That's handled by:

```go
	if pipelineVersion.Name != tmpl.V2PipelineName() {
		return nil, newBadRequestError("The object name must match spec.pipelineSpec.pipelineInformation.name")
	}
```